### PR TITLE
feat: create icon button component

### PR DESCRIPTION
**Pull request by nata-kab**
## Name PR according scheme: type of changes (feat/fix/refactor etc.) + description
feat: create IconButton

## Jira issue

JIRA issue: [FSU-93](https://klau.atlassian.net/browse/FSU-93)

## Describe your changes

Create IconButton with corresponding Storybook file and Jest tests.

## Checklist before requesting a review

- [x] Checked which branch you would merge.
- [x] Checked if the code follows the style guidelines of this project (use eslint and prettier).
- [x] Performed a self-review of code (bugs, clean code rules, etc).
- [x] Checked if added variables have unique names.
- [x] Checked if added code/function is not a duplicate. 
- [x] Tests have been added if possible.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: packages/design-system/components/iconButton/IconButton.styled.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import styled from "@emotion/styled"
+
+export const IconButton = styled.a`
+  display: flex;
```
Comment (paulgrym) [inline packages/design-system/components/iconButton/IconButton.styled.js:4]: I think inline-flex would fit better

**Review by paulgrym — COMMENTED** [review]
Looks really good - only one thing to think through and PR name to change because it doesn't pass 😅

**Comment by AmandaKr** [comment]
You need to change the name of the pr to feat: create icon button component  (all letters lowercase) to make PR pass

**Review by im-klau — COMMENTED** [review]
It's really difficult to find anything worth improving here, it's perfect! ❤️

File: packages/design-system/components/iconButton/IconButton.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { string, oneOf, func } from "prop-types"
+
+import Icon, { icons } from "design-system/components/icon"
+import theme from "../../theme/theme"
```
[the remark this answers is not in the document]
Comment (nata-kab) [reply packages/design-system/components/iconButton/IconButton.jsx]: 🫡  😀

**Review by im-klau — APPROVED** [review]

**Review by paulgrym — APPROVED** [review]
🚀

**Pull request merged**